osu.Game/Overlays/ChatOverlayV2.cs

```diff
@@ -248,13 +249,26 @@ private void currentChannelChanged(ValueChangedEvent<Channel> channel)
                 channelListing.State.Value = Visibility.Hidden;
                 textBar.ShowSearch.Value = false;
 
-                loading.Show();
-                LoadComponentAsync(new ChatOverlayDrawableChannel(newChannel), loaded =>
+                if (loadedChannels.ContainsKey(newChannel))
+                {
+                    currentChannelContainer.Clear(false);
+                    currentChannelContainer.Add(loadedChannels[newChannel]);
+                }
+                else
                 {
-                    currentChannelContainer.Clear();
-                    currentChannelContainer.Add(loaded);
-                    loading.Hide();
-                });
+                    loading.Show();
+
+                    // Ensure the drawable channel is stored before async load to prevent double loading
+                    ChatOverlayDrawableChannel drawableChannel = new ChatOverlayDrawableChannel(newChannel);
+                    loadedChannels.Add(newChannel, drawableChannel);
+
+                    LoadComponentAsync(drawableChannel, loaded =>
+                    {
+                        currentChannelContainer.Clear(false);
+                        currentChannelContainer.Add(loaded);
+                        loading.Hide();
+                    });
```

Comment:
Despite this already existing as-is, are you sure this works correctly? I would imagine it to have a conditional checking whether the current channel matches the latest channel joined (by `currentChannel.Value == channel.NewValue`).

Otherwise you could potentially have the wrong channel loaded, if the first channel takes a while to load:
```
currentChannelChanged(channel1)
channel1 loads...
currentChannelChanged(channel2)
channel2 loads...
channel2 finishes
channel1 finishes

expectation: channel2 shows up because it's the selected chnanel.
actual: channel1 shows up because it's the latest one to load.
```

I had also thought about using a cancellation token but that somewhat defeats the purpose of caching, because it cancels loading previous channels while we want to cache them in fully loaded state.

Reply:
Updated to ensure that the correct drawable channel is show after the async load